The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8216236882/job/22470583943) [(plain)](https://github.com/rust-lang/rust/commit/a1ad6ca80190eb4648e1f78093cd83d89b983a6a/checks/22470583943/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:b4ffde65f46336ab88eb53be808477a3936bae11)
Download action repository 'actions/upload-artifact@v3' (SHA:a8a3f3ad30e3422c9c7b888a15615d19a852ae32)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_93d140f6-66dd-4c13-801f-ecf319c8a435
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_PATH=/home/runner/work/_temp/_github_workflow/event.json
GITHUB_GRAPHQL_URL=https://api.github.com/graphql
GITHUB_HEAD_REF=check-dep-graph-size
GITHUB_JOB=pr
GITHUB_PATH=/home/runner/work/_temp/_runner_file_commands/add_path_93d140f6-66dd-4c13-801f-ecf319c8a435
GITHUB_REF=refs/pull/122245/merge
GITHUB_REF_NAME=122245/merge
GITHUB_REF_PROTECTED=false
---
COPY scripts/sccache.sh /scripts/
RUN sh /scripts/sccache.sh

COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt \
    && pip3 install virtualenv
COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---

#10 [5/8] COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
#10 DONE 0.0s

#11 [6/8] RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt     && pip3 install virtualenv
#11 0.679   Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
#11 0.691 Collecting boolean-py==4.0
#11 0.695   Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
#11 0.709 Collecting chardet==5.1.0
---
#11 3.667 Building wheels for collected packages: reuse
#11 3.668   Building wheel for reuse (pyproject.toml): started
#11 3.993   Building wheel for reuse (pyproject.toml): finished with status 'done'
#11 3.994   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#11 3.999   Stored in directory: /tmp/pip-ephem-wheel-cache-frikrstm/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#11 3.999 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#11 4.018   Attempting uninstall: setuptools
#11 4.018     Found existing installation: setuptools 59.6.0
#11 4.020     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#11 4.020     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#11 4.020     Can't uninstall 'setuptools'. No files were found to uninstall.
#11 4.685 Successfully installed binaryornot-0.4.4 boolean-py-4.0 chardet-5.1.0 jinja2-3.1.2 license-expression-30.0.0 markupsafe-2.1.1 python-debian-0.1.49 reuse-1.1.0 setuptools-66.0.0
#11 4.686 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#11 5.193 Collecting virtualenv
#11 5.233   Downloading virtualenv-20.25.1-py3-none-any.whl (3.8 MB)
#11 5.288      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.8/3.8 MB 71.1 MB/s eta 0:00:00
#11 5.335 Collecting platformdirs<5,>=3.9.1
#11 5.337   Downloading platformdirs-4.2.0-py3-none-any.whl (17 kB)
#11 5.355 Collecting distlib<1,>=0.3.7
#11 5.386   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#11 5.421 Collecting filelock<4,>=3.12.2
#11 5.424   Downloading filelock-3.13.1-py3-none-any.whl (11 kB)
#11 5.424   Downloading filelock-3.13.1-py3-none-any.whl (11 kB)
#11 5.508 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#11 5.669 Successfully installed distlib-0.3.8 filelock-3.13.1 platformdirs-4.2.0 virtualenv-20.25.1
#11 DONE 5.7s

#12 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#12 DONE 0.0s
---
DirectMap4k:      165824 kB
DirectMap2M:     6125568 kB
DirectMap1G:    12582912 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
    Finished dev [unoptimized] target(s) in 0.03s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/8401645716b26a8b4c6974dc0680e55e81e9e8a1/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-8401645716b26a8b4c6974dc0680e55e81e9e8a1-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished release [optimized] target(s) in 27.68s
##[endgroup]
fmt check
##[error]Diff in /checkout/compiler/rustc_incremental/src/persist/file_format.rs at line 18:
 use std::env;
 use std::io::{self, Read, Write};
-use std::path::{Path, PathBuf};
-use std::path::{Path, PathBuf};
 use std::io::{Seek, SeekFrom};
+use std::path::{Path, PathBuf};
 
 /// The first few bytes of files generated by incremental compilation.
 const FILE_MAGIC: &[u8] = b"RSIC";
Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/std/src/sys/pal/unix/thread_local_key.rs" "/checkout/library/std/src/sys/pal/unix/kernel_copy.rs" "/checkout/library/std/src/sys/pal/unix/stack_overflow.rs" "/checkout/library/std/src/sys/pal/unix/os.rs" "/checkout/library/std/src/sys/pal/unix/rand.rs" "/checkout/library/std/src/sys/pal/unix/weak.rs" "/checkout/compiler/rustc_incremental/src/persist/fs/tests.rs" "/checkout/library/std/src/sys/pal/unix/net.rs" "/checkout/library/std/src/sys/pal/unix/pipe.rs" "/checkout/library/std/src/sys/pal/unix/io.rs" "/checkout/compiler/rustc_incremental/src/persist/data.rs" "/checkout/compiler/rustc_incremental/src/persist/fs.rs" "/checkout/compiler/rustc_incremental/src/persist/file_format.rs" "/checkout/compiler/rustc_incremental/src/persist/mod.rs" "/checkout/compiler/rustc_incremental/src/persist/save.rs" "/checkout/compiler/rustc_incremental/src/persist/load.rs" "/checkout/compiler/rustc_incremental/src/persist/dirty_clean.rs" "/checkout/compiler/rustc_incremental/src/persist/work_product.rs" "/checkout/compiler/rustc_incremental/src/errors.rs" "/checkout/compiler/rustc_incremental/src/lib.rs" "/checkout/compiler/rustc_incremental/src/assert_dep_graph.rs" "/checkout/library/std/src/sys/pal/unix/kernel_copy/tests.rs" "/checkout/library/std/src/sys/pal/unix/l4re.rs" "/checkout/library/std/src/sys/pal/unix/stdio.rs" "/checkout/library/std/src/sys/pal/unix/args.rs" "/checkout/library/std/src/sys/pal/unix/alloc.rs" "/checkout/compiler/rustc_next_trait_solver/src/lib.rs" "/checkout/compiler/rustc_next_trait_solver/src/canonicalizer.rs" "/checkout/library/std/src/sys/pal/zkvm/env.rs" "/checkout/library/std/src/sys/pal/zkvm/abi.rs" "/checkout/library/std/src/sys/pal/zkvm/mod.rs" "/checkout/library/std/src/sys/pal/zkvm/thread_local_key.rs" "/checkout/library/std/src/sys/pal/zkvm/os.rs" "/checkout/library/std/src/sys/pal/zkvm/stdio.rs" "/checkout/library/std/src/sys/pal/zkvm/args.rs" "/checkout/library/std/src/sys/pal/zkvm/alloc.rs" "/checkout/compiler/rustc_graphviz/src/tests.rs" "/checkout/compiler/rustc_graphviz/src/lib.rs" "/checkout/library/std/src/sys/pal/wasip2/mod.rs" "/checkout/compiler/rustc_fluent_macro/src/fluent.rs" "/checkout/compiler/rustc_fluent_macro/src/lib.rs" "/checkout/library/std/src/sys/pal/solid/env.rs" "/checkout/library/std/src/sys/pal/solid/thread_local_dtor.rs" "/checkout/library/std/src/sys/pal/solid/error.rs" "/checkout/library/std/src/sys/pal/solid/time.rs" "/checkout/library/std/src/sys/pal/solid/fs.rs" "/checkout/library/std/src/sys/pal/solid/mod.rs" "/checkout/library/std/src/sys/pal/solid/thread_local_key.rs" "/checkout/library/std/src/sys/pal/solid/os.rs" "/checkout/library/std/src/sys/pal/solid/abi/fs.rs" "/checkout/library/std/src/sys/pal/solid/abi/mod.rs" "/checkout/library/std/src/sys/pal/solid/abi/sockets.rs" "/checkout/library/std/src/sys/pal/solid/net.rs" "/checkout/library/std/src/sys/pal/solid/io.rs" "/checkout/library/std/src/sys/pal/solid/stdio.rs" "/checkout/library/std/src/sys/pal/solid/alloc.rs" "/checkout/compiler/rustc_ast_pretty/src/pprust/state/expr.rs" "/checkout/compiler/rustc_ast_pretty/src/pprust/state/item.rs" "/checkout/compiler/rustc_ast_pretty/src/pprust/tests.rs" "/checkout/compiler/rustc_ast_pretty/src/pprust/mod.rs" "/checkout/compiler/rustc_ast_pretty/src/pprust/state.rs" "/checkout/compiler/rustc_ast_pretty/src/helpers.rs" "/checkout/compiler/rustc_ast_pretty/src/pp/convenience.rs" "/checkout/library/std/src/sys/pal/unix/os/tests.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Sat Mar  9 17:58:12 UTC 2024
  network time: Sat, 09 Mar 2024 17:58:12 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/compiler/rustc_query_system/src/dep_graph/serialized.rs
+++ b/compiler/rustc_query_system/src/dep_graph/serialized.rs
@@ -179,13 +179,15 @@ impl SerializedDepGraph {
     pub fn decode<D: Deps>(d: &mut MemDecoder<'_>) -> SerializedDepGraph {
         // The last 16 bytes are the node count and edge count.
         debug!("position: {:?}", d.position());
-        let (node_count, edge_count) =
-            d.with_position(d.len() - 2 * IntEncodedWithFixedSize::ENCODED_SIZE, |d| {
+        let (node_count, edge_count, graph_size) =
+            d.with_position(d.len() - 3 * IntEncodedWithFixedSize::ENCODED_SIZE, |d| {
                 debug!("position: {:?}", d.position());
                 let node_count = IntEncodedWithFixedSize::decode(d).0 as usize;
                 let edge_count = IntEncodedWithFixedSize::decode(d).0 as usize;
-                (node_count, edge_count)
+                let graph_size = IntEncodedWithFixedSize::decode(d).0 as usize;
+                (node_count, edge_count, graph_size)
             });
+        assert_eq!(d.len(), graph_size);
         debug!("position: {:?}", d.position());
 
         debug!(?node_count, ?edge_count);
@@ -491,6 +493,8 @@ impl<D: Deps> EncoderState<D> {
         debug!("position: {:?}", encoder.position());
         IntEncodedWithFixedSize(node_count).encode(&mut encoder);
         IntEncodedWithFixedSize(edge_count).encode(&mut encoder);
+        let graph_size = encoder.position() + IntEncodedWithFixedSize::ENCODED_SIZE;
+        IntEncodedWithFixedSize(graph_size as u64).encode(&mut encoder);
         debug!("position: {:?}", encoder.position());
         // Drop the encoder so that nothing is written after the counts.
         let result = encoder.finish();

